### PR TITLE
Unit testing

### PR DESCRIPTION
**Pull request by ixboy**
Hello, this is our unit tests for the To-Do List project, the requirements of this project were the following:

Use jest framework for testing.
Create a test file ([..].test.js) for a file containing the add item and delete item functions that you want to test.
Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be straightforward. Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks:

    Mock a storage object to "imitate" localStorage operations.
    Mock HTML to test if add/delete functions add or remove exactly one <li> element to/from the list in the DOM.

Make sure you group your tests using the description() method.

We have implemented all the requirements, please do review our project and tell us if we can proceed to the next one.

Thanks.

### CODE REVIEW

File: src/__tests__/add-delete.test.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+document.body.innerHTML = '<input type="text" placeholder="Add to your list..." id="new-task">';
+
+let toDoTasks = [];
+
+describe('My To-Do List', () => {
+  test('should add the new task into the list of to do list', () => {
+    const newTask = document.getElementById('new-task');
+    newTask.value = 'Wash the dishes';
+    toDoTasks = addNewTaskToList();
```
Comment (DeliceLydia) [inline src/__tests__/add-delete.test.js:18]: - [x] Please fix one test which is failing
test

- [ ] Consider testing positive and negative scenarios

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! However, there are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._



### Cheers and Happy coding!👏👏👏

Feel free to leave a question, or a comment on the PR thread if something is not clear 100%. 

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/__tests__/add-delete.test.js (hunk 1 above)
Comment (sinansevgi) [reply src/__tests__/add-delete.test.js:18]: [Optional]
As suggested by the previous TSE, it is always best practice to test your application for negative cases also.

**Review by sinansevgi — APPROVED** [review]
# Approved :heavy_check_mark: 

Hi @ixboy and @Maha-Magdy , 

All test cases are passing. You have tested all required functions.
Your project is complete!  
Congratulations! 🎉

You can merge this pull request and continue with your journey :wave: 


Happy Hacking! :computer: 

-----

You can reply to this conversation if you have any questions or need elaboration. :wave: 
Sinan Sevgi


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**